presto-hive-metastore/src/main/java/com/facebook/presto/hive/PartitionVersionFetcher.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import org.apache.hadoop.hive.metastore.api.Partition;
+
+import java.util.Optional;
+
+public interface PartitionVersionFetcher
+{
+    Optional<Integer> getPartitionVersion(Partition partition);
```

Comment:
Not high priority for this PR, but might worth thinking about how to handle open partitions (where data is still actively being ingested to the partition, internally known as NRT). These partitions might not be cacheable across queries, especially partition stats.

Reply:
This is a very good point !. Looks like we have a enum in the FB Partition metadata that tells us if the partition in APPENDABLE or FINALIZED. Will send out another PR to fix the caching in such case.

Reply:
Yeah, we should not cache the in progress partitions.

Reply:
Looks like we could just skip providing versions for in progress partitions so we would always treat it as stale :D